Bottom: acb90057a5c2227df4d480223275c85609da35d2
Top:    da1e70cd8fc10de520644a8dad2aa12fe55f82c4
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-29 23:44:38 +0100

DrawLayer(): Pass clear_pins argument as true when calling DrawLayerCommon

This should not affect any rendering. If the check planes flag is set, we
should only draw polygons, nothing more. Only a handful of cases draw layers
via this code-path, and in all but the exporter cases, they are not called
if CHECKPLANES is set.

The exporters which call DrawLayer explicitly clear the CHECKPLANES flag
(since commit c00ee74844cad14ecd55bae854a82ff6412633e6), so nothing should
be affected by this argument change.


---

diff --git a/src/draw.c b/src/draw.c
index 83080b3..fb29826 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -1113,7 +1113,7 @@ DrawLayerCommon (LayerTypePtr Layer, const BoxType * screen, bool clear_pins)
 void
 DrawLayer (LayerTypePtr Layer, const BoxType * screen)
 {
-  DrawLayerCommon (Layer, screen, false);
+  DrawLayerCommon (Layer, screen, true);
 }
 
 /* ---------------------------------------------------------------------------
